application/forms/IcingaDependencyForm.php

```diff
@@ -212,11 +212,18 @@ protected function addObjectsElement()
 
         if (!empty($sentParent) || $dependency->isApplyRule()) {
             $parentService = $dependency->get('parent_service');
+            if ($parentService === null) {
+                $parentServiceVar = $dependency->get('parent_service_by_name') ?? '';
+                if (strlen($parentServiceVar) > 0) {
+                    $parentService = '$' . $dependency->get('parent_service_by_name') . '$';
```

Comment:
* Drop the null coalesce operation
* Don't use strlen if the actual length is not of interest to you
* Use the variable instead of accessing the property again